File: Parth Mathur/maxSubArray.java
class Solution {
  public int maxSubArray(int[] nums) {
      int currsum=0;
      int maxsum= Integer.MIN_VALUE;
      if(nums.length==1)
          return nums[0];
      for(int i=0;i<nums.length;i++){
          currsum +=nums[i];
          if(currsum>maxsum)
              maxsum=currsum;
          if(currsum<0){
              currsum=0;
          }
      }
      return maxsum;
  }
}

File: Mudit Choraria/max-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ans = nums[0];
        int curr = 0;
        for(int i: nums) {
            curr = max(curr + i, i);
            ans = max(curr, ans);
        }
        return ans;
    }
};
